src/browser_action/vitals.js

```diff
@@ -224,53 +233,145 @@
     );
   }
 
-  function addUserTimings(metric) {
+  function addUserTimings(metric, enableUserTiming, enableConsoleTables) {
     switch (metric.name) {
       case "LCP":
-        // LCP has a loadTime/renderTime (startTime), but not a duration.
-        // Could visualize relative to timeOrigin, or from loadTime -> renderTime.
-        // Skip for now.
+        if (metric.attribution && metric.attribution.lcpEntry && metric.attribution.navigationEntry) {
+          const lcpEntry = metric.attribution.lcpEntry;
+          const navEntry = metric.attribution.navigationEntry;
+          // Set the start time to the later of the actual start time or the activationStart (for prerender) or 0
+          const startTime = Math.max(navEntry.startTime, navEntry.activationStart) || 0;
+          // Add the performance marks for the Performance Panel
+          if (enableUserTiming) {
+              performance.measure(`[Web Vitals Extension] LCP.timeToFirstByte`, {
+              start: startTime,
+              duration: lcpEntry.timeToFirstByte,
+            });
+            performance.measure(`[Web Vitals Extension] LCP.resourceLoadDelay`, {
+              start: startTime + lcpEntry.timeToFirstByte,
```

Comment:
Getting this error:

> Uncaught (in promise) TypeError: Failed to execute 'measure' on 'Performance': Failed to read the 'start' property from 'PerformanceMeasureOptions': The provided double value is non-finite.
 
Seems like the TTFB value isn't part of the `lcpEntry` object: `metric.attribution.timeToFirstByte`. I think this applies to all of the LCP diagnostic measures.

Reply:
I can't repeat this, and TTFB is part of metric.attribution.timeToFirstByte as shown with the console.log entry:

<img width="324" alt="image" src="https://user-images.githubusercontent.com/10931297/232163422-397275a5-e1dd-40b3-b149-61bb9fad3a19.png">

I recall having a similar error on an older version which I fixed but struggling to remember in which commit. Can you check you're looking at the latest version on this branch?

Reply:
Right, the bug is that the correct path is `metric.attribution.timeToFirstByte` but you're attempting to access it from `metric.attribution.lcpEntry.timeToFirstByte`.

This is the first time I've pulled from your branch, so it should be the latest version. Are you able to see the LCP timings in the Perf panel?

Reply:
Ah see it now. Was looking at console.log. Fixed now.